src/ZPublisher/HTTPRequest.py

```diff
@@ -1204,11 +1204,11 @@ def resolve_url(self, url):
         # method is called on the response).
         try:
             object = req.traverse(path)
-        except Exception:
+        except Exception as exc:
             rsp.exception()
-        if object is None:
-            req.clear()
-            raise sys.exc_info()[0](rsp.errmsg)
+            if object is None:
+                req.clear()
+                raise exc.__class__(rsp.errmsg)
```

Comment:
If the `req.clear()` and `raise` are supposed to happen inside the `except` block, then the `if object is None:` test is bogus.